src/main/kotlin/no/nav/familie/ef/sak/journalføring/JournalføringService.kt

```diff
@@ -117,26 +114,54 @@ class JournalføringService(
             vilkårsbehandleNyeBarn = journalføringRequest.vilkårsbehandleNyeBarn
         )
 
-        if (journalpost.journalstatus != Journalstatus.JOURNALFOERT) {
-            journalpostService.oppdaterJournalpostMedFagsakOgDokumenttitler(
-                journalpost,
-                journalføringRequest.dokumentTitler,
-                fagsak.eksternId.id,
-                saksbehandler
-            )
-            journalpostService.ferdigstillJournalføring(
-                journalpost.journalpostId,
-                journalføringRequest.journalførendeEnhet,
-                saksbehandler
-            )
-        }
+        journalpostService.oppdaterOgFerdigstillJournalpost(
+            journalpost = journalpost,
+            dokumenttitler = journalføringRequest.dokumentTitler,
+            journalførendeEnhet = journalføringRequest.journalførendeEnhet,
+            fagsak = fagsak,
+            saksbehandler = saksbehandler
+        )
 
         ferdigstillJournalføringsoppgave(journalføringRequest)
         opprettBehandlingsstatistikkTask(behandling.id, journalføringRequest.oppgaveId.toLong())
 
         return opprettSaksbehandlingsoppgave(behandling, saksbehandler)
     }
 
+    fun automatiskJournalførTilFørstegangsbehandling(
+        journalpostId: String,
+        personIdent: String,
+        stønadstype: StønadType
+    ): AutomatiskJournalføringResponse {
+        val fagsak = fagsakService.hentEllerOpprettFagsak(personIdent, stønadstype)
+        val journalpost = journalpostService.hentJournalpost(journalpostId)
+        val journalførendeEnhet = arbeidsfordelingService.hentNavEnhetIdEllerBrukMaskinellEnhetHvisNull(fagsak.hentAktivIdent())
+        infotrygdPeriodeValideringService.validerKanJournalføresGittInfotrygdData(fagsak)
```

Comment:
bytt ut denne med "totalsjekken" for kanOppretteFørstegangsbehandlingINyLøsning...?